keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py

```diff
@@ -314,18 +319,59 @@ def is_installed(force_recheck=False):
     return installation_status(force_recheck)[0]
 
 
-def _do_pkt_shadow_copy():
-    import tempfile
-    import shutil
+def _remove_dir(dir):
+    try:
+        shutil.rmtree(dir, ignore_errors=True)
+    except Exception as err:
+        logger = logging.getLogger(__name__)
+        logger.error('remove_dir: Cannot delete dir {}'.format(dir))
+        logger.error('Exception info: {}'.format(str(err)))
+
+
+def _get_all_pids_on_windows():
+    output = subprocess.run(['tasklist', '/NH', '/FO', 'CSV'],
+                            capture_output=True)  # windows specific code
+    rows = re.split(b'\n', output.stdout)
+    pids = []
+    for row in rows:
+        elems = re.split(b'","', row)
+        try:
+            pid = int(elems[1])
+            pids.append(pid)
+        except Exception:
+            pass
+    return pids
+
+
+def _all_dirs_in_dir(dir):
+    return [f for f in os.listdir(dir) if os.path.isdir(os.path.join(dir, f))]
+
+
+def _remove_old_dirs(base_dir):
+    pids = _get_all_pids_on_windows()
+    dirs = _all_dirs_in_dir(base_dir)
```

Comment:
swap pids= and dirs=  lines here. 
```
pids = []
# new process `1` starts and creates dir
dir = ['1'] # this dir gets removed
```
False positive is a bug here, false negative is OK

Reply:
I agree. Fixed